tests/fixtures/energy_dashboard.py

```diff
@@ -0,0 +1,357 @@
+        {"source": "Machinery", "target": "Carbon Dioxide", "value": "1.0"},
+        {
+            "source": "Oil and Gas Processing",
+            "target": "Carbon Dioxide",
+            "value": "3.6",
+        },
+        {"source": "Oil and Gas Processing", "target": "Methane", "value": "2.8"},
+        {"source": "Other Agriculture", "target": "Methane", "value": "1.4"},
+        {"source": "Other Agriculture", "target": "Nitrous Oxide", "value": "0.3"},
+        {
+            "source": "Other Fuel Combustion",
+            "target": "Agricultural Energy Use",
+            "value": "1.0",
+        },
+        {
+            "source": "Other Fuel Combustion",
+            "target": "Commercial Buildings",
+            "value": "1.3",
+        },
+        {
+            "source": "Other Fuel Combustion",
+            "target": "Residential Buildings",
+            "value": "5.0",
+        },
+        {
+            "source": "Other Fuel Combustion",
+            "target": "Unallocated Fuel Combustion",
+            "value": "1.8",
+        },
+        {"source": "Other Industry", "target": "Carbon Dioxide", "value": "6.6"},
+        {"source": "Other Industry", "target": "HFCs - PFCs", "value": "0.4"},
+        {
+            "source": "Pulp - Paper and Printing",
+            "target": "Carbon Dioxide",
+            "value": "1.1",
+        },
+        {
+            "source": "Rail - Ship and Other Transport",
+            "target": "Carbon Dioxide",
+            "value": "2.5",
+        },
+        {
+            "source": "Residential Buildings",
+            "target": "Carbon Dioxide",
+            "value": "10.2",
+        },
+        {"source": "Rice Cultivation", "target": "Methane", "value": "1.5"},
+        {"source": "Road", "target": "Carbon Dioxide", "value": "10.5"},
+        {"source": "T and D Losses", "target": "Carbon Dioxide", "value": "2.2"},
+        {"source": "Transportation", "target": "Air", "value": "1.7"},
+        {
+            "source": "Transportation",
+            "target": "Rail - Ship and Other Transport",
+            "value": "2.5",
+        },
+        {"source": "Transportation", "target": "Road", "value": "10.5"},
+        {
+            "source": "Unallocated Fuel Combustion",
+            "target": "Carbon Dioxide",
+            "value": "3.0",
+        },
+        {"source": "Unallocated Fuel Combustion", "target": "Methane", "value": "0.4"},
+        {
+            "source": "Unallocated Fuel Combustion",
+            "target": "Nitrous Oxide",
+            "value": "0.4",
+        },
+        {"source": "Waste", "target": "Landfills", "value": "1.7"},
+        {"source": "Waste", "target": "Waste water - Other Waste", "value": "1.5"},
+        {"source": "Waste water - Other Waste", "target": "Methane", "value": "1.2"},
+        {
+            "source": "Waste water - Other Waste",
+            "target": "Nitrous Oxide",
+            "value": "0.3",
+        },
+    ]
```

Comment:
I wonder if we should utilize the existing data fetching logic from the examples data? See `superset/examples/helpers.py` and `superset/examples/energy.py:load_energy()`. I think we could potentially cache the data on disk, so it only needs to be fetched once if it's already there (I don't expect this data to change that often).

Reply:
@villebro to be honest, I imagine as developer who wants write/run tests (especially unit tests), I would like have data that isn't fetched from web - so I don't need internet to just run tests. Actually this is why I didn't use this fetching on purpose. Also if want use fixtures, probably we would like to cleanup after each test. I think it would extend test setup significantly then. But what do you think about just moving it to separate file (as energy_usage.json) or just shortening collection (if it's possible)?

Actually I had an idea to generate this data with some random values of energy sources, targets and values. Apparently it requires some structure (as far as I remember it complained about some cycles). Maybe you know how this data should be structured so I could generate it with loop?

Reply:
@kkucharc regarding having fixtures that are generated as opposed to fetched, I do agree that it makes sense. Some time ago I added some `DataFrame` fixtures (before `pytest`, hence they're not done the elegant way):

https://github.com/apache/incubator-superset/blob/ac498ddd2310f1692d6bf204042a795bd5bcf897/tests/fixtures/dataframes.py#L106-L116

This and others in this file are used to test some `DataFrame` transformation functions in `utils/pandas_postprocessing.py`, where I can aggregate/sort/filter the `DataFrame` and then assert that the result is as expected. What makes this convenient is that the dataset is very simple, yet provides enough complexity to be able to be able to ensure that the functions work as expected.

So, for this case, perhaps we should just completely move off the energy datasets, and start using some programmatically generated dataset, and then start applying that template to all other examples datasets that are used throughout the backend tests.

@dpgaspar @betodealmeida @hughhhh do you have thoughts about how to proceed here?

Reply:
My 2 cents would be to migrate away from the energy test fixture entirely and move to random data. I agree with Kasha's statement that relying on a data load from the network isn't a great thing for a test suite. I'd prefer for us to move to testing with more-generic and randomly generated data.